content/language/php/run-tests.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Run PHP tests in a container
+keywords: php, test
+description: Learn how to run your PHP tests in a container.
+---
+
+## Prerequisites
+
+Complete all the previous sections of this guide, starting with [Containerize a PHP application](containerize.md).
+
+## Overview
+
+Testing is an essential part of modern software development. Testing can mean a
+lot of things to different development teams. There are unit tests, integration
+tests and end-to-end testing. In this guide you take a look at running your unit
+tests in Docker when developing and when building.
+
+## Run tests when developing locally
+
+The sample application already has a PHPUnit test inside the `tests` directory. When developing locally, you can use Compose to run your tests.
+
+Run the following command in the `docker-php-sample` directory to run the tests inside a container.
+
+```console
+$ docker compose run --build --rm server ./vendor/bin/phpunit tests/HelloWorldTest.php
+```
+
+You should see output that contains the following.
+
+```console
+Hello, Docker!PHPUnit 9.6.13 by Sebastian Bergmann and contributors.
+
+.                                                                   1 / 1 (100%)
+
+Time: 00:00.003, Memory: 4.00 MB
+
+OK (1 test, 1 assertion)
+```
+
+To learn more about the command, see [docker compose run](/engine/reference/commandline/compose_run/).
+
+## Run tests when building
+
+To run your tests when building, you need to update your Dockerfile. Create a new test stage that runs the tests.
+
+The following is the updated Dockerfile.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+FROM composer:lts as prod-deps
+WORKDIR /app
+RUN --mount=type=bind,source=./composer.json,target=composer.json \
+    --mount=type=bind,source=./composer.lock,target=composer.lock \
+    --mount=type=cache,target=/tmp/cache \
+    composer install --no-dev --no-interaction
+
+FROM composer:lts as dev-deps
+WORKDIR /app
+RUN --mount=type=bind,source=./composer.json,target=composer.json \
+    --mount=type=bind,source=./composer.lock,target=composer.lock \
+    --mount=type=cache,target=/tmp/cache \
+    composer install --no-interaction
+
+FROM php:8.2-apache as base
+RUN docker-php-ext-install pdo pdo_mysql
+COPY ./src /var/www/html
+
+FROM base as development
+COPY ./tests /var/www/html/tests
+RUN mv "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"
+COPY --from=dev-deps app/vendor/ /var/www/html/vendor
+
+FROM development as test
+WORKDIR /var/www/html
+RUN ./vendor/bin/phpunit tests/HelloWorldTest.php
+
+FROM base as final
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+COPY --from=prod-deps app/vendor/ /var/www/html/vendor
+USER www-data
```

Comment:
I think you received a comment about moving the "test" stage to before the final stage to not override the default. As an alternative solution, we could add the test stage to the end of the Dockerfile, but also add a final "FROM final" line to force `final` to be the default stage.

```dockerfile
FROM development as test
WORKDIR /var/www/html
RUN ./vendor/bin/phpunit tests/HelloWorldTest.php

FROM final
```

I think this could help make this instruction more clear. ("Add the following to your Dockerfile")

Reply:
I think I'll add the highlighted lines here and leave the test stage above final. Otherwise, I need to briefly explain the last line, which might take focus away from the main objective.